# HG changeset patch
# User Rik <rik@octave.org>
# Date 1517609790 28800
#      Fri Feb 02 14:16:30 2018 -0800
# Node ID aa9c29f48718fbd3952c7fbe0be41169fafbf880
# Parent  69b89e81d3d3cb3f60d109ab60be45f687a30ab2
configure.ac: Only check for psapi library when compiling for Windows.

* configure.ac: Check host_os before calling AC_CHECK_LIB([psapi]).

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1522,16 +1522,23 @@ case $host_os in
                    [AC_MSG_ERROR([Missing function required to build Octave])])
 
     ## We need this for GetDeviceCaps.  Should we actually check for
     ## the function and library?
     LIBS="-lgdi32 $LIBS"
   ;;
 esac
 
+## Windows-specific module used for winqueryreg function
+case $host_os in
+  msdosmsvc | mingw*)
+    AC_CHECK_LIB([psapi], [EnumProcessModules])
+  ;;
+esac
+
 ## Cygwin kluge for getrusage.
 AC_CHECK_FUNCS([getrusage])
 case $host_os in
   cygwin*)
     AC_DEFINE(RUSAGE_TIMES_ONLY, 1,
       [Define to 1 if the struct rusage only has time information.])
   ;;
 esac
@@ -1973,19 +1980,16 @@ if test "$have_x" = yes; then
     AC_CHECK_LIB([X11], XrmInitialize, [X11_LIBS="-lX11"], [X11_LIBS=])
   elif test "$x_libraries" != "NONE"; then
     AC_CHECK_LIB([X11], XrmInitialize,
       [X11_LIBS="-L$x_libraries -lX11"], [X11_LIBS=], "-L$x_libraries")
   fi
   AC_SUBST(X11_LIBS)
 fi
 
-## FIXME: This check should probably be for Windows only.
-AC_CHECK_LIB([psapi], [EnumProcessModules])
-
 ### Check for the Carbon framework on macOS systems
 
 OCTAVE_HAVE_FRAMEWORK([Carbon],
   [[#include <Carbon/Carbon.h>]], [[CGMainDisplayID ()]],
   [have_framework_carbon=yes], [have_framework_carbon=no])
 if test $have_framework_carbon = yes; then
   AC_DEFINE(HAVE_FRAMEWORK_CARBON, 1,
     [Define to 1 if framework CARBON is available.])
